File: CSharp/5kyu/CountIPAddresses.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CSharp._5kyu
{
    public static partial class Kata5
    {
        public static long IpsBetween(string start, string end)
        {
            return IpValue(end) - IpValue(start);
        }

        public static long IpValue(string ip)
        {
            var result = ip.Split(".").Select((b, i) => Convert.ToInt64(b) << 24 - (8 * i));
            return result.Sum();
        }
    }
}

File: Python/5kyu/CountIPAddresses.py
def ips_between(start, end):
    return ip_value(end) - ip_value(start)


def ip_value(ip):
    return sum(int(b) << 24 - 8 * i for i, b in enumerate(ip.split(".")))


print(ips_between("20.0.0.0", "20.0.0.50"))
